### PR TITLE
DT-1903: 🐛 Lowercase emails from nomis and change all existing

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/OidcJwtAuthenticationSuccessHandler.kt [hunk 1]
```diff
@@ -72,13 +72,13 @@ class OidcJwtAuthenticationSuccessHandler(
...
 
     return AzureUserPersonDetails(
-      ArrayList(),
-      true,
-      principal.getClaim<String>("oid").toUpperCase(),
-      givenName,
-      familyName,
-      principal.preferredUsername.toLowerCase(),
-      true,
+      authorities = ArrayList(),
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/OidcJwtAuthenticationSuccessHandler.kt:75]: no change in functionality, just changing to use parameter names for constructor so clearer which fields are being set to what

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/verify/VerifyEmailService.kt [hunk 2]
```diff
@@ -44,23 +42,13 @@ class VerifyEmailService(
 
   fun getExistingEmailAddressesForUsername(username: String): List<String> =
     jdbcTemplate.queryForList(EXISTING_EMAIL_SQL, mapOf("username" to username), String::class.java)
+      .map { it.toLowerCase() }
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/verify/VerifyEmailService.kt:45]: lowercase emails from database

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/verify/VerifyEmailService.kt [hunk 3]
```diff
@@ -44,23 +42,13 @@ class VerifyEmailService(
...
-      emailsByUsername
-        .computeIfAbsent(username) { HashSet() }
-        .add(email)
-    }
-    return emailsByUsername
+  fun getExistingEmailAddressesForUsernames(usernames: List<String>): Map<String, Set<String>> {
+    if (usernames.isEmpty()) return emptyMap()
+
+    return jdbcTemplate.query(EXISTING_EMAIL_FOR_USERNAMES_SQL, mapOf("usernames" to usernames)) { rs: ResultSet, _: Int -> rs.getString("USERNAME") to rs.getString("EMAIL") }
+      .groupBy { it.first }.mapValues { it.value.map { p -> p.second.toLowerCase() }.toSet() }
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/verify/VerifyEmailService.kt:51]: change horrible mutable map into kotlin goodness

File: src/main/resources/db/auth/V5_34__user_email_lowercase.sql [hunk 4]
```diff
@@ -0,0 +1,3 @@
+update users
+set email = lower(email)
```
Comment (petergphillips) [inline src/main/resources/db/auth/V5_34__user_email_lowercase.sql:2]: change all existing rows in database (>6000 in prod) to be lowercase

File: src/main/resources/db/dev/data/nomis/V9_1__user_data.sql [hunk 5]
```diff
@@ -420,7 +420,7 @@ Insert into internet_addresses (INTERNET_ADDRESS_ID, OWNER_CLASS, OWNER_ID, OWNE
                                 MODIFY_DATETIME, MODIFY_USER_ID, AUDIT_TIMESTAMP, AUDIT_USER_ID, AUDIT_MODULE_NAME,
                                 AUDIT_CLIENT_USER_ID, AUDIT_CLIENT_IP_ADDRESS, AUDIT_CLIENT_WORKSTATION_NAME,
                                 AUDIT_ADDITIONAL_INFO)
-VALUES (138250, 'STF', 4, null, null, 'EMAIL', 'phillips@bobjustice.gov.uk',
+VALUES (138250, 'STF', 4, null, null, 'EMAIL', 'Phillips@BobJustice.gov.uk',
```
Comment (petergphillips) [inline src/main/resources/db/dev/data/nomis/V9_1__user_data.sql:423]: test data to ensure that emails are lowercased - tests fail without the lowercase changes

**Review by mikehalmamoj — APPROVED** [review]

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**